webrender/src/border.rs

```diff
@@ -894,9 +965,33 @@ impl BorderRenderTaskInfo {
         Some(BorderRenderTaskInfo {
             border_segments,
             size: size.to_i32(),
+            available_size_dependence: size_dependence,
         })
     }
 
+    /// Returns the cache key size for this task, based on our available size
+    /// dependence computed earlier.
+    #[inline]
+    pub fn cache_key_size(
+        &self,
+        local_size: &LayoutSize,
+        scale: LayoutToDeviceScale,
+    ) -> DeviceIntSize {
+        let mut size = DeviceIntSize::zero();
+        if self.available_size_dependence.is_empty() {
+            return size;
```

Comment:
why aren't we returning the `*local_size * scale` here?

Reply:
Because for borders where we don't depend on any size we just want the same value for all of them, so they hit the cache and we don't build instances unnecessarily.

Reply:
what about a particular width/height then? shouldn't we put 0.0 into width if the key is only height-sensitive?

Reply:
That's what we do.

Reply:
Note that we return `size` instead of `device_size`. I'm open to suggestions about how to make that clear.